### ISSUE TITLE
Some directories are not processed and built even though they have all requirements.

### ISSUE THREAD

**Issue by madsem** [11ty/eleventy#1867]
**Describe the bug**
Unsure if it's a bug or if i'm doing something entirely wrong here.

I have created a dir structure in my `_site` dir that matches the recommended structure for Netlify CMS with i18n enabled.

I'm using permalinks and data files to redirect the output to directories recommended for 11ty with 18n setup.

The goal here is to be able to manage content via Netlify CMS in multiple languages, and deploy it so that 11ty language switcher etc will work.

But, some directories are simply not processed, the files are nowhere to be found in my output folder.
I've enabled DEBUG, it simply shows messages that the particular files were found, and **added to map**. 

But after that there are no other log entries for these files, the built completes successfully without any errors and some files/folders are found and have been built as expected, others are completely ignored.

**To Reproduce**
Here is a repo, as you can see i have included the dist dir, and only a few dirs / files actually made it into there.
Everything should be in there to reproduce it.

https://github.com/madsem/11ty-issue

Please let me know if I can in any way help, would love to be able to get this site done haha :)

PS:

Maybe I should elaborate on the structure a bit more.

The `site/categories` directory is meant to contain the homepages for categories (and maybe also to build categories somehow). The files should be built into their respective collection folders though, as index files.

i.e: `site/categories/de/articles.md` should become `dist/de/articles/index.html`

This is kind of how most of the stuff in there is set up, for example the `home` dir as well.

**Comment by madsem** [comment]
@pdehaan I'm sorry! :D I just pushed it.

**Comment by madsem** [comment]
Maybe connected to #1783?

**Comment by pdehaan** [comment]
I'm a bit lost.

1. You have https://github.com/madsem/11ty-issue/blob/main/site/categories/en/articles.md, but it isn't generating any files in the output directory, which I found a bit suspicious.
    1. There is a https://github.com/madsem/11ty-issue/blob/main/site/categories/en/en.json file, but it seems to se the `dir` and `locale` frontmatter for all the files in the site/categories/en/** files.
    2. There is a https://github.com/madsem/11ty-issue/blob/main/site/categories/categories.json file which seems to set the `layout` and `permalink` frontmatter properties for all the site/categories/** files (maybe&hellip; I think you need to set [`config.setDataDeepMerge(true);`](https://www.11ty.dev/docs/data-deep-merge/) in your .eleventy.js file).
    3. There is a https://github.com/madsem/11ty-issue/blob/main/site/_includes/layouts/category.njk layout file, which seems to set the parent layout and uses pagination to loop over all collections (except "collections.all" and "collections.blog" and do something.

I added a debug page somewhere in the root and dumped out all the collection names using this custom filter in .eleventy.js:

```js
config.addFilter("keys", (data) => Object.keys(data).sort());
```
```njk
{{ collections | keys }}
<!-- Output: "all" -->
```

So if I'm following all the code correctly, you need to set the data deep merge to true (I believe this will be the default in Eleventy v1), so all the directory data files get merged, and then you need some tags/collections somewhere. It looks like the site only has a single collection, "all", which is filtered out of the pagination loop in ./site/_includes/layouts/category.njk, which means there is nothing to loop over, which means no category pages get written). I created a new file named "bananas.njk" somewhere in the project and added `tags: ["bananas"]` in the frontmatter, and then it created a bunch more files.

It might need a bit more debugging. I had to fiddle with the `permalink` structure on a page or two to get it to avoid naming collisions, but that might have been a bug on my part since I was doing a bit of debugging and maybe messed up one of your original values.

**Comment by madsem** [comment]
wow thanks a lot, so it seems there's more going on than I thought. I will have a look at it right in the am :)

Is the way i'm doing it there a little unusual as to how 11ty usually works? Or why are you lost?

**Comment by madsem** [comment]
I'm trying your recommendations, so far no real changes in behavior though.

I've enabled dataDeepMerge, set tags in all dir data files. The root dirs for collections are built now, with an empty index.html. I have also deleted the contents of the `category.njk`, because this is actually a remnant of the theme I used to built from. (I should have probably mentioned that yesterday).

The content that's inside the index.md files is entirely missing.

I'm wondering if my way of structuring the project is at fault here, or if there's something not working as expected.

**Comment by pdehaan** [comment]
You might need to push your changes to GitHub. I'm not 100% clear on where you're adding the tags, and I don't seem to be seeing empty HTML files.

If you're missing content in your layout files, make sure you have that `{{ content }}` somewhere in there.

Here's the changes I made to site/_includes/layouts/category.njk (which you may or may not have deleted):

```diff
diff --git a/site/_includes/layouts/category.njk b/site/_includes/layouts/category.njk
index 48acb22..46ad8c3 100644
--- a/site/_includes/layouts/category.njk
+++ b/site/_includes/layouts/category.njk
@@ -7,7 +7,7 @@ pagination:
   filter:
     - all
     - blog
-permalink: /categories/{{ category }}/
+permalink: "{{ locale }}/categories/{{ category }}/{{ page.fileSlug }}/"
 hide_from_sitemap: true
 ---

@@ -17,4 +17,8 @@ hide_from_sitemap: true
   {% endfor %}
 </ul>

+<pre>
+{{ content | safe }}
+</pre>
+
 <p class="text-400">Back to <a href="/categories/">all categories</a></p>
```

And now my generated dist/de/categories/home/articles/index.html file has the following content:

```html
<ul class="category-home leading-loose">
  <li><a href="/de/">German Homepage</a></li>
  <li><a href="/">Homepage</a></li>
  <li><a href="/es/">Spanish Homepage</a></li>
  <li><a href="/fr/">French Homepage</a></li>
</ul>

<pre>
<p>This, is how we roll!</p>
</pre>

<p class="text-400">Back to <a href="/categories/">all categories</a></p>
```

(where I added `"tags": ["home"]` to the site/home/home.json file)

```diff
diff --git a/site/home/home.json b/site/home/home.json
index a417369..90a3dc9 100644
--- a/site/home/home.json
+++ b/site/home/home.json
@@ -1,5 +1,6 @@
 {
     "layout": "home",
-    "permalink": "{{ locale }}/"
+    "permalink": "{{ locale }}/",
+    "tags": ["home"]
   }
```

**Comment by madsem** [comment]
@pdehaan I got it all solved, so it was def. not an 11ty issue but simply insufficient knowledge on my part :)

Got it all solved with your help though. The issue was indeed that the category.njk  was still in the layouts dir.
I thought if I don't reference it it cannot cause issues, which I now know was wrong :)

In the end it all turned out exactly the way I envisioned, I can now have my dir structure in `site` so that Netlify CMS can read/write to it, and when building the site I can completely change the dir structure!
11ty is really extremely flexible, it just takes a moment until you grasp how it all works.

I can even use the Netlify `relation` widget to use the `slug` as linking field, then dynamically group related content in the output folder based on the related slug, like so:
```js
module.exports = {
  layout: "page",
  tags: ["review"],
  permalink: "{{ locale }}/{{ related_widget_slug }}/reviews/{{ page.fileSlug }}/",
};
```

I'm asking myself how did I ever built sites without 11ty haha. 

Thanks again for all your input, that was a great help.
